javascript/lifecycle.js

```diff
@@ -125,17 +126,29 @@ document.addEventListener(
 // - element - the element that triggered the reflex (not necessarily the Stimulus controller's element)
 //
 export const dispatchLifecycleEvent = (stage, element, reflexId) => {
+  const { reflexData, reflexController } = element
+
+  if (!document.body.contains(element)) {
+    const attrs = extractElementAttributes(element)
+    element = findElement(attrs, true)
+    element.reflexData = reflexData
+    element.reflexController = reflexController
+  }
+
   if (!element) return
   if (!element.reflexData) element.reflexData = {}
   if (!element.reflexController) element.reflexController = {}
+
   const { target } = element.reflexData[reflexId] || {}
+  const { controller } = element.reflexController[reflexId] || {}
+
   element.dispatchEvent(
     new CustomEvent(`stimulus-reflex:${stage}`, {
       bubbles: true,
       cancelable: false,
       detail: {
         reflex: target,
-        controller: element.reflexController[reflexId],
+        controller,
```

Comment:
I already really like what you did here; it's what I should have done initially. Thanks for having my back.